src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -982,14 +980,7 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof SSLCallback) {
                     SSLCallback sslCallback = (SSLCallback) callback;
-
-                    try {
-                        X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
-                        verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
```

Comment:
But you are correct: the verify step cannot be skipped. In my previous comment I tried to make that clear... the client cert should be verified, but the cert field must be set from server certain.